stdlib/public/core/ContiguousCollection.swift

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// A collection that supports access to its underlying contiguous storage.
+@available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+public protocol ContiguousCollection: Collection
+where SubSequence: ContiguousCollection {
+  /// Calls a closure with a pointer to the array's contiguous storage.
+  func withUnsafeBufferPointer<R>(
+    _ body: (UnsafeBufferPointer<Element>) throws -> R
+  ) rethrows -> R
+}
+
+@available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+public extension ContiguousCollection {
+  @inlinable
+  func withUnsafeBufferPointer<R>(
+    _ body: (UnsafeBufferPointer<Element>) throws -> R
+  ) rethrows -> R {
+    return try withContiguousStorageIfAvailable(body)!
+  }
+}
+
+/// A collection that supports mutable access to its underlying contiguous
+/// storage.
+@available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+public protocol MutableContiguousCollection: ContiguousCollection, MutableCollection
+where SubSequence: MutableContiguousCollection {
+  /// Calls the given closure with a pointer to the array's mutable contiguous
```

Comment:
```suggestion
  /// Calls the given closure with a pointer to mutable contiguous
```

Remove: `the array's` (also in the proposal).